src/internet_identity_interface/src/lib.rs

```diff
@@ -173,3 +173,92 @@ pub struct HttpResponse {
 pub struct InternetIdentityInit {
     pub assigned_user_number_range: (UserNumber, UserNumber),
 }
+
+// Archive specific types
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub enum OperationType {
+    #[serde(rename = "register_anchor")]
+    RegisterAnchor {
+        initial_device: DeviceDataWithoutAlias,
+    },
+    #[serde(rename = "add_device")]
+    AddDevice { new_device: DeviceDataWithoutAlias },
+    #[serde(rename = "update_device")]
+    UpdateDevice {
+        updated_device: PublicKey,
+        changed_data: DeviceDataUpdate,
+    },
+    #[serde(rename = "remove_device")]
+    RemoveDevice { removed_device: PublicKey },
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub struct LogEntry {
```

Comment:
nit: now that the canister is called "archive", should this just be an "entry"? Same for `Logs` and where applicable